content/doc/book/pipeline/syntax.adoc

```diff
@@ -97,6 +97,21 @@ section is placed. The section must be defined at the top-level inside the
 | In the top-level `pipeline` block and each `stage` block.
 |===
 
+[[differences-between-top-and-stage-level]]
+===== Differences between top and stage level Agents
+
+There are some nuances when adding an agent to the top level or a stage level, and this when the `options` directive is applied.
+[[top-level-agents]]
+====== Top Level Agents
+In top level `agents` options are invoked after entering in the agent. As example, when using `timeout` it will be only
+applied to the execution within the agent.
+
+[[stage-level-agents]]
+====== Stage Agents
+In stage level `agents` options are invoked before entering the `agent` or checking any `when` conditions. In this case,
```

Comment:
Sentence per line and a few minor proposed changes.
```suggestion
In `agents` declared within a stage, the options are invoked *before* entering the `agent` or checking any `when` conditions.
In this case, when using `timeout`, it is applied *before* the `agent` is allocated.
The timeout will include the agent provisioning time.
Because the timeout includes the agent provisioning time, the Pipeline may fail in cases where agent allocation is delayed.
```